modules/core/src/matrix_transform.cpp

```diff
@@ -516,92 +517,119 @@ flipHoriz( const uchar* src, size_t sstep, uchar* dst, size_t dstep, Size size,
-                int t1 = ((int*)(src1 + i))[0];
+                for( ; i <= sz.width - 16; i += 16 )
+                {
+                    int t0 = ((int*)(srcTop + i))[0];
+                    int t1 = ((int*)(srcBottom + i))[0];
 
-                ((int*)(dst0 + i))[0] = t1;
-                ((int*)(dst1 + i))[0] = t0;
+                    ((int*)(dstTop + i))[0] = t1;
+                    ((int*)(dstBottom + i))[0] = t0;
 
-                t0 = ((int*)(src0 + i))[1];
-                t1 = ((int*)(src1 + i))[1];
+                    t0 = ((int*)(srcTop + i))[1];
+                    t1 = ((int*)(srcBottom + i))[1];
 
-                ((int*)(dst0 + i))[1] = t1;
-                ((int*)(dst1 + i))[1] = t0;
+                    ((int*)(dstTop + i))[1] = t1;
+                    ((int*)(dstBottom + i))[1] = t0;
 
-                t0 = ((int*)(src0 + i))[2];
-                t1 = ((int*)(src1 + i))[2];
+                    t0 = ((int*)(srcTop + i))[2];
+                    t1 = ((int*)(srcBottom + i))[2];
 
-                ((int*)(dst0 + i))[2] = t1;
-                ((int*)(dst1 + i))[2] = t0;
+                    ((int*)(dstTop + i))[2] = t1;
+                    ((int*)(dstBottom + i))[2] = t0;
 
-                t0 = ((int*)(src0 + i))[3];
-                t1 = ((int*)(src1 + i))[3];
+                    t0 = ((int*)(srcTop + i))[3];
+                    t1 = ((int*)(srcBottom + i))[3];
 
-                ((int*)(dst0 + i))[3] = t1;
-                ((int*)(dst1 + i))[3] = t0;
-            }
+                    ((int*)(dstTop + i))[3] = t1;
+                    ((int*)(dstBottom + i))[3] = t0;
+                }
 
-            for( ; i <= size.width - 4; i += 4 )
-            {
-                int t0 = ((int*)(src0 + i))[0];
-                int t1 = ((int*)(src1 + i))[0];
+                for( ; i <= sz.width - 4; i += 4 )
+                {
+                    int t0 = ((int*)(srcTop + i))[0];
+                    int t1 = ((int*)(srcBottom + i))[0];
 
-                ((int*)(dst0 + i))[0] = t1;
-                ((int*)(dst1 + i))[0] = t0;
+                    ((int*)(dstTop + i))[0] = t1;
+                    ((int*)(dstBottom + i))[0] = t0;
+                }
             }
-        }
 
-        for( ; i < size.width; i++ )
-        {
-            uchar t0 = src0[i];
-            uchar t1 = src1[i];
+            for( ; i < sz.width; i++ )
+            {
+                uchar t0 = srcTop[i];
+                uchar t1 = srcBottom[i];
 
-            dst0[i] = t1;
-            dst1[i] = t0;
+                dstTop[i] = t1;
+                dstBottom[i] = t0;
+            }
         }
     }
+private:
+    const uchar* src0_;
+    size_t sstep_;
+    uchar* dst0_;
+    size_t dstep_;
+    Size size_;
+    size_t esz_;
+    int stripe_;
+};
+
+static void
+flipVert( const uchar* src0, size_t sstep, uchar* dst0, size_t dstep, Size size, size_t esz )
+{
+    int nthreads = getNumberOfCPUs();
```

Comment:
Add `const` to both nthreads and stripe.

Reply:
modified accordingly, thanks